packages/api/src/external/carequality/document/process-outbound-document-query-resps.ts

```diff
@@ -147,44 +130,15 @@ export async function processOutboundDocumentQueryResps({
     // Since we have most of the document contents when doing the document query,
     // we will store this in FHIR and then upsert the reference to the s3 object in FHIR
     // when doing the doc retrieval
+    const docRefsWithMetriportId = filterDocRefsWithMetriportId(docsToDownload);
     await storeInitDocRefInFHIR(docRefsWithMetriportId, cxId, patientId, log);
 
-    const resultsWithMetriportIdAndDrUrl: OutboundDocumentQueryResp[] = [];
-
-    const replaceDqUrlWithDrUrl = async (
-      outboundDocumentQueryResp: OutboundDocumentQueryResp
-    ): Promise<void> => {
-      const gateway = await getCQDirectoryEntry(outboundDocumentQueryResp.gateway.homeCommunityId);
-
-      if (!gateway) {
-        const msg = `Gateway not found - Doc Retrieval`;
-        log(`${msg}: ${outboundDocumentQueryResp.gateway.homeCommunityId} skipping...`);
-        capture.message(msg, {
-          extra: {
-            context: `cq.dq.getCQDirectoryEntry`,
-            patientId,
-            requestId,
-            cxId,
-            gateway: outboundDocumentQueryResp.gateway,
-          },
-        });
-        return;
-      } else if (!gateway.urlDR) {
-        log(`Gateway ${gateway.id} has no DR URL, skipping...`);
-        return;
-      }
-
-      resultsWithMetriportIdAndDrUrl.push({
-        ...outboundDocumentQueryResp,
-        gateway: {
-          ...outboundDocumentQueryResp.gateway,
-          url: gateway.urlDR,
-        },
-      });
-    };
-
-    await executeAsynchronously(resultsWithMetriportId, replaceDqUrlWithDrUrl, {
```

Comment:
The reason for the bug was because we were referring to `resultsWithMetriportId` here but this was not filtered for docs that were already downloaded